pkg/controllers/plugin/helm_controller.go

```diff
@@ -105,14 +106,48 @@ func (r *HelmReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 	pluginStatus := initPluginStatus(plugin)
 	defer func() {
 		if statusErr := setPluginStatus(ctx, r.Client, plugin, pluginStatus); statusErr != nil {
-			log.FromContext(ctx).Error(statusErr, "failed to set status")
+			logger.Error(statusErr, "failed to set status")
 		}
 	}()
 
-	clusterAccessReadyCondition, restClientGetter := initClientGetter(ctx, r.Client, r.kubeClientOpts, *plugin)
+	var restClientGetter genericclioptions.RESTClientGetter
+	clusterAccessReadyCondition := greenhousev1alpha1.UnknownCondition(greenhousev1alpha1.ClusterAccessReadyCondition, "", "")
+
+	cluster := &greenhousev1alpha1.Cluster{}
+	if plugin.Spec.ClusterName != "" {
+		err := r.Client.Get(ctx, types.NamespacedName{Namespace: plugin.GetNamespace(), Name: plugin.Spec.ClusterName}, cluster)
+		if err != nil {
+			clusterAccessReadyCondition.Status = metav1.ConditionFalse
+			clusterAccessReadyCondition.Message = fmt.Sprintf("Failed to get cluster %s: %s", plugin.Spec.ClusterName, err.Error())
+			pluginStatus.StatusConditions.SetConditions(clusterAccessReadyCondition)
+			return ctrl.Result{}, err
+		}
+	}
+
+	clusterAccessReadyCondition, restClientGetter = initClientGetter(ctx, r.Client, r.kubeClientOpts, *plugin, cluster, clusterAccessReadyCondition)
 	pluginStatus.StatusConditions.SetConditions(clusterAccessReadyCondition)
+	// check if plugin is ready to be reconciled
+	// if the cluster is marked for deletion or other error during initClientGetter, the condition is false
+	// if deletionTimeStamp is set, the deletion logic will continue as long as the cluster is not scheduled for deletion
 	if !clusterAccessReadyCondition.IsTrue() {
-		return ctrl.Result{}, fmt.Errorf("cannot access cluster: %s", clusterAccessReadyCondition.Message)
+		// if the cluster is scheduled for deletion, we should not reconcile the plugin
+		// but rather requeueAfter with duration diff of currentTime and schedule time
+		// ClusterDeletionScheduledReason is only set if the cluster is scheduled for deletion
+		if clusterAccessReadyCondition.Reason == greenhousev1alpha1.ClusterDeletionScheduledReason {
```

Comment:
The `clusterAccessReadyCondition` is the one on the `Plugin` here, right?

We should check `ClusterDeletionScheduled` on the cluster here and probably not the `ClusterDeletionScheduledReason`?

Reply:
done